Userland/Libraries/LibVideo/PlaybackManager.cpp

```diff
@@ -96,7 +114,10 @@ Time PlaybackManager::current_playback_time()
 
 Time PlaybackManager::duration()
 {
-    auto duration_result = m_demuxer->duration();
+    auto duration_result = ({
+        auto demuxer_locker = Threading::MutexLocker(m_demuxer_mutex);
+        m_demuxer->duration();
+    });
```

Comment:
Yeow, expression statement in normal code? I would rather this be a lambda that you invoke immediately with a FIXME to port the `with_locked`/`MutexProtected` pattern from the Kernel to LibThreading.

Reply:
Not opposed to this being an immediately-invoked lambda, but is there a difference in behavior between the two here? The only difference I've observed is in initialization of constexpr variables where only a lambda will work.